package-res/resources/web/pentaho/data/_doc/IAbstractFilter.jsdoc

```diff
@@ -0,0 +1,85 @@
+/*!
+ * Copyright 2010 - 2015 Pentaho Corporation.  All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @ignore
+ * @classdesc The `IAbstractFilter` interface allows creation of other types of filters for a data table.
+ *
+ * @name pentaho.data.filter.IAbstractFilter
+ * @interface
+ */
+
+/**
+ * Tests if an entry is an element of the set defined by this filter.
+ *
+ * @name pentaho.data.filter.AbstractFilter#contains
+ * @method
+ * @abstract
+ * @param {pentaho.data.filter._Element} - [dataTable]{@link pentaho.data.Table} entry.
+ * @return {boolean}
+ */
+
+/**
+ * Returns a filter that is the inverse of this filter.
+ *
+ * @name pentaho.data.filter.AbstractFilter#invert
+ * @method
+ * @abstract
+ * @returns {*}
+ */
+
+/**
+ * Returns a filter that is the union of this filter with another.
+ * In other words, implements the OR operation between this filter and another.
+ *
+ * @name pentaho.data.filter.AbstractFilter#or
+ * @method
+ * @abstract
+ * @param {} other -
+ * @returns {*}
+ */
+
+/**
+ * Returns a filter that is the intersection of this filter with another.
+ * In other words, implements the INTERSECT operation between this filter and another.
+ *
+ * @name pentaho.data.filter.AbstractFilter#and
+ * @method
+ * @abstract
+ * @param {} other -
+ * @returns {*}
+ */
+
+/**
+ * Applies the filters to the data table
+ *
+ * @name pentaho.data.filter.AbstractFilter#apply
+ * @method
+ * @abstract
+ * @param {pentaho.data.Table} dataTable The data table to filter
+ * @returns {*}
+ */
+
+/**
+ * Outputs a simple object that serializes the operation described by this filter.
+ * The syntax loosely follows the query language of MongoDB.
+ *
+ * @name pentaho.data.filter.AbstractFilter#toSpec
+ * @method
+ * @abstract
```

Comment:
Please add an @ignore, as this functionality is not really part of this story and we haven't settled that this is the way we want to serialize a filter